go/arrow/avro/schema.go

```diff
@@ -0,0 +1,407 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package avro reads Avro OCF files and presents the extracted data as records
+
+package avro
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/apache/arrow/go/v14/internal/types"
+	avro "github.com/hamba/avro/v2"
+)
+
+type schemaNode struct {
+	name         string
+	parent       *schemaNode
+	schema       avro.Schema
+	union        bool
+	children     []*schemaNode
+	arrowField   arrow.Field
+	path         []string
+	index, depth int32
+}
+
+var schemaCache avro.SchemaCache
```

Comment:
this should probably not be global to the package, otherwise multiple calls to read schemas might get cached schemas that aren't correct....?

Reply:
ok, I think the only way to not have a global variable would be to pass a pointer to the schema cache to the schemaNode object and all its children; will do that.